docs/en/sql-reference/functions/type-conversion-functions.md

```diff
@@ -49,105 +49,1378 @@ SETTINGS cast_keep_nullable = 1
 └──────────────────┴─────────────────────┴──────────────────┘
 ```
 
-## toInt(8\|16\|32\|64\|128\|256)
+## toInt8
 
-Converts an input value to a value the [Int](../data-types/int-uint.md) data type. This function family includes:
+Converts an input value to a value of type [`Int8`](../data-types/int-uint.md). Throws an exception in case of an error.
 
-- `toInt8(expr)` — Converts to a value of data type `Int8`.
-- `toInt16(expr)` — Converts to a value of data type `Int16`.
-- `toInt32(expr)` — Converts to a value of data type `Int32`.
-- `toInt64(expr)` — Converts to a value of data type `Int64`.
-- `toInt128(expr)` — Converts to a value of data type `Int128`.
-- `toInt256(expr)` — Converts to a value of data type `Int256`.
+**Syntax**
+
+```sql
+toInt8(expr)
+```
+
+**Arguments**
+
+- `expr` — Expression returning a number or a string representation of a number. [Expression](../syntax.md/#syntax-expressions).
+
+Supported types:
+- (U)Int8/16/32/64/128/256
+- Float*
+- String representations of (U)Int8/16/32/128/256
+
+Unsupported types:
+- Float values `NaN` and `Inf` throw an exception.
+- String representations of binary and hexadecimal values, e.g. `SELECT toInt8('0xc0fe');`
+
+:::note
+If the input value cannot be represented within the bounds of [Int8](../data-types/int-uint.md), the result over or under flows. This is not considered an error.  
```

Comment:
I am not a native English speaker but `over or under flows` looks grammatically incorrect (sorry, I proposed to write it that way).

Can we say `the result overflows (or underflows)` instead?

Reply:
For new docs, could we have one line / sentence (https://sembr.org/)?

Reply:
> I am not a native English speaker but `over or under flows` looks grammatically incorrect (sorry, I proposed to write it that way).
> 
> Can we say `the result overflows (or underflows)` instead?

I went with "If the input value cannot be represented within the bounds of Int*, overflow or underflow of the result occurs. This is not considered an error."